### DISCUSSION TITLE
restore just one app's data manually - ie: via ADB restore

### DISCUSSION THREAD

**Discussion by JohnTheLearner** [seedvault-app/seedvault#547] [Q&A] [answered]
Manually backup data for only a single app from a seedvault backup...maybe using ADB restore?  I don't want to restore the whole SV backup or any of the system parameters...just data for a single app.  

I've tried restoring using ADB using just one app's directory of data taken from a SV backup and re-packed into an adb backup file as shown in https://stackpointer.io/mobile/android-adb-backup-extract-restore-repack/372/.  The OS prompts, runs the restore, but seems to not take effect in the app...even after a restart.  TIA.

**Comment by JohnTheLearner** [comment]
So to clarify, is there a way to manipulate the backup found in `./SeedVaultAndroidBackup` to remove system artifacts and other apps not desired to restore, then re-crypt the backup such that one could restore that on new hardware?  I understand this isn't a typical use case and would be more of a manual process.  I See the new, v1, extractor, but extracts-only and I don't see an encrypt option to re-form a backup once the above modifications are complete. TIA and feel free to PM if preferred.

**Reply by chirayudesai** [reply, marked as answer]
The manual process here would involve having a tool which is able to re-crypt the backup. we do have the seed (which can be used to generate the key), but such a tool isn't available right now.

Another way would be to put the backup into the typical `adb backup/restore` format, not sure if that'll still work on latest Android though.

In the future, SeedVault itself will allow you to selectively restore.
